### ISSUE TITLE
Split search term by whitespaces

### ISSUE THREAD

**Issue by dinkom** [mlavin/django-selectable#149]
Hi.

I've made some adjustments to the get_query() method in the ModelLookup class:

```
def get_query(self, request, term):
        qs = self.get_queryset()
        if term:
            search_filters = []
            if self.search_fields:
                for t in term.split():
                    for field in self.search_fields:
                        search_filters.append(Q(**{field: t}))
            qs = qs.filter(reduce(operator.or_, search_filters))
        return qs
```

It allows users to enter more terms in one go, for instance, name & surname. It's a bit slower than the original, obviously, but not that much from what I've noticed.

Could something like this find it's way into the package? If you think it could, I could write it better (keep the old behaviour as default, but add an option for this).

Cheers,
Dinko M.

**Comment by mlavin** [comment]
I think something like this could be generally useful and not just for the model lookups. Perhaps the lookup classes need a way to split the search string into search terms. Not just to split on whitespace but also to potentially remove common/stop words.

**Comment by dinkom** [comment]
I could try writing something like that perhaps later this week and then we can review it. Do you agree?

**Comment by mlavin** [comment]
By all means go for it.

**Comment by PetrDlouhy** [comment]
@dinkom: I have written to-character exactly the same code before I found this :-)

@mlavin I tried to fulfill also your requirements in #192, so please review.

**Comment by vkurup** [comment]
I think this can be closed thanks to @PetrDlouhy's PR in #192.

**Comment by mlavin** [comment]
:space_invader:
